x-pack/legacy/plugins/code/server/lib/esqueue/worker.js

```diff
@@ -465,7 +466,11 @@ export class Worker extends events.EventEmitter {
         if (jobs.length > 0) {
           this.debug(`${jobs.length} outstanding jobs returned`);
         }
-        return jobs;
+        return jobs.filter(async job => {
```

Comment:
I probably don't have the full context, but which problem do we want to solve here?